fl4health/utils/metrics.py

```diff
@@ -133,140 +260,51 @@ def __call__(self, logits: torch.Tensor, target: torch.Tensor) -> Scalar:
-            self.metric_values_history[i].append(metric_value)
+        Class to manage a set of metrics associated to a given prediction type.
 
-    def compute(self) -> Dict[str, Scalar]:
-        """
-        Returns average of each metrics given its historical values and counts
+        Args:
+            metrics (Sequence[Metric]): List of metric to evaluate predictions on.
+            metric_mngr_name (str): Name of the metric manager (ie train, val, test)
         """
-        total_count = sum(self.counts)
-        weights: List[float] = [count / total_count for count in self.counts]
-
-        metric_value_averages = []
-        for metric_values in self.metric_values_history:
-            avg = sum([weight * float(val) for weight, val in zip(weights, metric_values)])
-            metric_value_averages.append(avg)
-
-        results: Dict[str, Scalar] = {
-            f"{self.name}_{str(metric)}".lstrip("_"): avg for metric, avg in zip(self.metrics, metric_value_averages)
-        }
-
-        return results
-
-    def clear(self) -> None:
-        self.metric_values_history = [[] for _ in range(len(self.metrics))]
-        self.counts = []
+        self.og_metrics = metrics
+        self.metric_mngr_name = metric_mngr_name
+        self.metrics_per_prediction_type: Dict[str, Sequence[Metric]] = {}
 
+    def update(self, preds: Dict[str, torch.Tensor], target: torch.Tensor) -> None:
+        """
+        Updates (or creates then updates) a list of metrics for each prediction type.
 
-class MetricMeterManager:
-    """
-    Class to manage one or metric meters.
-    """
+        Args:
+            preds (Dict[str, torch.Tensor]): A dictionairy of
+        """
+        if len(self.metrics_per_prediction_type) == 0:
+            self.metrics_per_prediction_type = {key: copy.deepcopy(self.og_metrics) for key in preds.keys()}
 
-    def __init__(self, key_to_meter_map: Dict[str, MetricMeter]):
-        self.key_to_meter_map = key_to_meter_map
+        for pred, mtrcs in zip(preds.values(), self.metrics_per_prediction_type.values()):
+            for m in mtrcs:
+                m.update(pred, target)
 
-    def update(self, preds: Dict[str, torch.Tensor], target: torch.Tensor) -> None:
-        for pred_key in preds.keys():
-            if pred_key in self.key_to_meter_map.keys():
-                self.key_to_meter_map[pred_key].update(preds[pred_key], target)
+    def compute(self) -> Metrics:
+        """
+        Computes set of metrics for each prediction type.
 
-    def compute(self) -> Dict[str, Scalar]:
+        Returns:
+            Metrics: Dictionairy containing computed metrics along with string identifiers
+                for each prediction type.
+        """
         all_results = {}
-        for meter in self.key_to_meter_map.values():
-            result = meter.compute()
-            all_results.update(result)
+        for metrics_key, mtrcs in self.metrics_per_prediction_type.items():
```

Comment:
Super minor, but let's avoid the abbreviation here, since `metrics` is pretty short anyway.